cartoframes/io/managers/context_manager.py

```diff
@@ -294,12 +315,38 @@ def _truncate_table(self, table_name, schema, cartodbfy):
 
     def _truncate_and_drop_add_columns(self, table_name, schema, df_columns, table_columns, cartodbfy):
         log.debug('TRUNCATE AND DROP + ADD columns table "{}"'.format(table_name))
-        query = '{regenerate}; BEGIN; {truncate}; {drop_columns}; {add_columns}; {cartodbfy}; COMMIT;'.format(
+        drop_columns = _drop_columns_query(table_name, table_columns)
+        add_columns = _add_columns_query(table_name, df_columns)
+
+        drop_add_columns = 'ALTER TABLE {table_name} {drop_columns},{add_columns};'.format(
+            table_name=table_name, drop_columns=drop_columns, add_columns=add_columns)
+
+        query = '{regenerate}; BEGIN; {truncate}; {drop_add_columns}; {cartodbfy}; COMMIT;'.format(
             regenerate=_regenerate_table_query(table_name, schema) if self._check_regenerate_table_exists() else '',
             truncate=_truncate_table_query(table_name),
-            drop_columns=_drop_columns_query(table_name, table_columns),
-            add_columns=_add_columns_query(table_name, df_columns),
+            drop_add_columns=drop_add_columns,
             cartodbfy=_cartodbfy_query(table_name, schema) if cartodbfy else '')
+
+        if len(query) > BATCH_API_PAYLOAD_THRESHOLD:
```

Comment:
I would add it in this way to make the two cases a bit more readable:

```py
if len(query) < BATCH_API_PAYLOAD_THRESHOLD:
    query = ...
else:
    query = ...
```

Reply:
Mmm at that point query wouldn't be defined yet 🤔 , do you think it'd be worth it to change it to something similar to the following or leave it as is?

```python
query = ...
if len(query) < BATCH_API_PAYLOAD_THRESHOLD:
    pass
else:
    query = ...
```

Reply:
Oh, you're right XD

It's OK as it is.